yourgov/components/FormMobileFoodVendorPermit/task1/FormTask1Step1ChangeDetails.tsx

```diff
@@ -53,91 +53,86 @@ export function FormTask1Step1ChangeDetails() {
 				...formState,
 				task1: { ...formState.task1, step1: { ...data } },
 			});
-			router.push(step1Path);
+			router.push(`${step1Path}?success=true`);
 		}, 1500);
 	};
 
-	function onDiscardChangesClick() {
+	function onCancelClick() {
 		router.push(step1Path);
 	}
 
-	const onError: SubmitErrorHandler<Task1Step1FormSchema> = () => {
-		setFocusedError(false);
-	};
-
-	// Only show the page alert if there is more than 1 error
-	const hasErrors = Object.keys(errors).length > 1;
-
-	useEffect(() => {
-		if (hasErrors && !focusedError) {
-			errorRef.current?.focus();
-			setFocusedError(true);
-		}
-	}, [hasErrors, focusedError, errors]);
+	const showErrorAlert = hasMultipleErrors(errors);
 
 	const titleRef = useRef<HTMLHeadingElement>(null);
 
 	// Focus the title when the user navigates to this page
 	// This usually happens in `FormContainer` component but we can't use that component here
+	// FIXME: Supposedly because of Progress Indicator? I think we can workaround that?
 	useEffect(() => {
 		titleRef.current?.focus();
 	}, []);
 
+	const { pathname } = useRouter();
+	const { canConfirmAndSubmit } = useFormTask1Context();
+
+	function getStepStatus(stepIndex: number): ProgressIndicatorItemStatus {
+		const step = task1FormSteps[stepIndex];
+		// Current step is always in progress when the URL matches
+		if (step.href === pathname.replace('/change-details', '')) return 'started';
+		// After submitting each step, the `completed` key is set to `true`
+		if (formState.task1?.[step.formStateKey]?.completed) return 'done';
+		// The final step (confirm and submit) can only be viewed when all previous steps are complete
+		if (step.formStateKey === 'step7' && !canConfirmAndSubmit) return 'blocked';
+		// Otherwise, the step still needs to be done
+		return 'todo';
+	}
+
 	return (
 		<Columns>
-			<Column columnSpan={{ xs: 12, md: 8 }}>
+			<Column columnSpan={{ xs: 12, md: 4, lg: 3 }}>
+				<ProgressIndicator
+					activePath={`${
+						task1FormSteps[0].items && task1FormSteps[0].items[0]?.href
```

Comment:
This may evaluate to `'undefined'`, but that probably won't happen with our controlled thing and doesn't really matter for the purposes of the demo - is that right?

Reply:
Again, only to appease TS. We've hardcoded the form steps array